[docs] Get started docs

jonsamp opened the pull request:
# Why

Adds getting started docs that match our new default template. My goal is to present a small selection of docs that can get anyone who'd like to start developing an Expo app started. Another goal is to keep these docs succinct, defaulting to showing users information rather than explaining it in text.

Adds:
- /get-started/introduction
- /get-started/create-a-project
- /get-started/set-up-your-environment
- /get-started/start-developing
- /get-started/next-steps

# Test plan

Look at the preview docs site and read the docs. Use them to set up a new project + development environment. I'd love feedback about how they feel to be read + used.

jonsamp committed:
allow mdx fragments

ide commented on docs/pages/get-started/set-up-your-environment.mdx:
```diff
@@ -0,0 +1,31 @@
+
+## How would you like to develop?
+
+Expo Go is a sandbox for trying out Expo quickly. A development build is a build of your app that includes Expo's developer tools.
```
"own" here helps add more emphasis about the difference:
```suggestion
Expo Go is a sandbox for trying out Expo quickly. A development build is a build of your own app that includes Expo's developer tools.
```

amandeepmittal commented on docs/pages/get-started/create-a-project.mdx:
```diff
@@ -1,117 +1,22 @@
+- macOS, Windows (Powershell and WSL 2), and Linux are supported.
 
-## Initialize a new project
+We recommend creating a project with the default project template, which includes example code to help get you started. To create a new project, run the following command:
```
```suggestion
We recommend creating a project with `create-expo-app` that sets up a default project template and includes example code to help get you started. To create a new project, run the following command:
```

- Adding a little bit of context about what `create-expo-app` helps telling what it does (which is the context already added here).

jonsamp replied:
I am including this comment with the following:

```
We recommend starting with the template project created by `create-expo-app`. This project includes example code to help you get started.

To create a new project, run the following command:
```

Again, and with all my rewordings, please lmk if you prefer something else!

amandeepmittal replied:
This is perfect! Thanks!

amandeepmittal commented on docs/pages/get-started/introduction.mdx:
```diff
@@ -0,0 +1,27 @@
+<BoxLink
+  title="Quick start"
+  description="Read the docs to create a project and set up your development environment."
+  href="/get-started/create-a-project"
```
```suggestion
  href="/get-started/create-a-project/"
```

jonsamp replied:
Oh! I didnt know we needed trailing slashes

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/PlatformAndDeviceForm.tsx:
```diff
@@ -0,0 +1,87 @@
+        <SelectCard
+          imgSrc="/static/images/get-started/android-emulator.png"
+          darkImgSrc="/static/images/get-started/android-emulator-dark.png"
+          title="Android emulator"
```
```suggestion
          title="Android Emulator"
```

When we use "Android" as a prefix, we tend to follow same pattern as Android documentation does is to capitalize "Emulator". So let's apply the same pattern here to be concistent.

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/PlatformAndDeviceForm.tsx:
```diff
@@ -0,0 +1,87 @@
+        <SelectCard
+          imgSrc="/static/images/get-started/ios-simulator.png"
+          darkImgSrc="/static/images/get-started/ios-simulator-dark.png"
+          title="iOS simulator"
```
```suggestion
          title="iOS Simulator"
```

When we use "iOS" as a prefix before simulator, we tend to follow same pattern as Apple documentation does is to capitalize "Simulator". So let's apply the same pattern here to be consistent (we also have a guideline for this in Vale which throws a warning here).

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidPhysicalExpoGo.mdx:
```diff
@@ -0,0 +1,12 @@
+
+## Android device with Expo Go
+
+Scan the QR code to download the app from the Google Play Store, or visit the Expo Go page on [the Google Play Store](https://play.google.com/store/apps/details?id=host.exp.exponent&referrer=docs).
```
```suggestion
Scan the QR code to download the app from the Google Play Store, or visit the Expo Go page on the [Google Play Store](https://play.google.com/store/apps/details?id=host.exp.exponent&referrer=docs).
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/iosPhysicalExpoGo.mdx:
```diff
@@ -0,0 +1,9 @@
+
+## iOS device with a Expo Go
+
+Scan the QR code to download the app from the App Store, or visit the Expo Go page on [the App Store](https://itunes.apple.com/app/apple-store/id982107779).
```
```suggestion
Scan the QR code to download the app from the App Store, or visit the Expo Go page on the [App Store](https://itunes.apple.com/app/apple-store/id982107779).
```

amandeepmittal commented on docs/pages/get-started/set-up-your-environment.mdx:
```diff
@@ -0,0 +1,31 @@
+
+<DevelopmentEnvironmentInstructions />
+
+## Next steps
```
```suggestion
## Next step
```

amandeepmittal commented on docs/pages/get-started/start-developing.mdx:
```diff
@@ -0,0 +1,109 @@
+
+After running the command above, you will see a QR code in your terminal. You can scan this QR code to open the app on your device.
+
+If you're using an Android emulator or iOS simulator, you can press <kbd>a</kbd> or <kbd>i</kbd> respectively to open the app.
```
```suggestion
If you're using an Android Emulator or iOS Simulator, you can press <kbd>a</kbd> or <kbd>i</kbd> respectively to open the app.
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidSimulatedDevelopmentBuild.mdx:
```diff
@@ -0,0 +1,58 @@
+import AndroidStudioInstructions from './_androidStudioInstructions.mdx';
+import AndroidEmulatorInstructions from './_androidEmulatorInstructions.mdx';
+
+## Set up an Android emulator with a development build
```
```suggestion
## Set up an Android Emulator with a development build
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidSimulatedExpoGo.mdx:
```diff
@@ -0,0 +1,11 @@
+import AndroidStudioInstructions from './_androidStudioInstructions.mdx';
+import AndroidEmulatorInstructions from './_androidEmulatorInstructions.mdx';
+
+## Set up an Android emulator with Expo Go
```
```suggestion
## Set up an Android Emulator with Expo Go
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidSimulatedExpoGo.mdx:
```diff
@@ -0,0 +1,11 @@
+
+## Install Expo Go
+
+When you start a development server with `npx expo start` on the [start developing](/get-started/start-developing) page, press <kbd>a</kbd> to open the Android emulator. Expo CLI will install Expo Go automatically.
```
```suggestion
When you start a development server with `npx expo start` on the [start developing](/get-started/start-developing) page, press <kbd>a</kbd> to open the Android Emulator. Expo CLI will install Expo Go automatically.
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/iosSimulatedExpoGo.mdx:
```diff
@@ -0,0 +1,9 @@
+
+## Install Expo Go
+
+When you start a development server with `npx expo start` on the [start developing](/get-started/start-developing) page, press <kbd>i</kbd> to open the iOS simulator. Expo CLI will install Expo Go automatically.
```
```suggestion
When you start a development server with `npx expo start` on the [start developing](/get-started/start-developing) page, press <kbd>i</kbd> to open the iOS Simulator. Expo CLI will install Expo Go automatically.
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,169 @@
+import { Terminal } from '~/ui/components/Snippet';
+import { Tabs, Tab } from '~/ui/components/Tabs';
+
+## Install Node.js, Watchman, and JDK
```
```suggestion
## Install Watchman and JDK
```

We should avoid listing Node.js as an installation step here since this is covered in the previous page under system requirements: http://docs.expo.dev-pr-28511.s3-website-us-east-1.amazonaws.com/get-started/create-a-project/.

Originally, this instruction worked in a standalone guide but not a good experience from user's pov to find out on the next page (after reading system requirement) we are telling them to install Node.js with a version management tool. 

If we want to be a bit opinionated about it, we can say use a version management tool under system requirement or avoid it altogether.

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,169 @@
+
+#### Install dependencies
+
+<Step label="1">
+
+Install the LTS release of Node.js using a [version management tool](https://nodejs.org/en/download/package-manager) such as `nvm` or `volta`. If you have already installed Node.js on your system, make sure the version is 16 or above.
```
Same as above comment. We can remove this.

byCedric replied:
Also, let's not recommend Node versions that are end of life (16) 😄

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,169 @@
+
+<Tab label="Linux">
+
+To set up your development environment on Linux for Android, you will install Node.js, Watchman, JDK, and Android Studio.
```
```suggestion
To set up your development environment on Linux for Android, you will install Watchman, JDK, and Android Studio.
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,169 @@
+
+#### Install dependencies
+
+<Step label="1">
+
+Install Node.js 18 or above by following [installation instructions for your Linux distribution](https://nodejs.org/en/download/package-manager).
+
+</Step>
```
Same as previous comment about duplicated Node.js info.

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,169 @@
+
+<Tab label='Windows'>
+
+To set up your development environment on Windows for Android, you will install Node.js, JDK, and Android Studio.
```
```suggestion
To set up your development environment on Windows for Android, you will install Node.js, JDK, and Android Studio.
```
```suggestion
To set up your development environment on Windows for Android, you will install JDK, and Android Studio.
```

Same as previous comment about duplicated Node.js info.

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,169 @@
+
+#### Install dependencies
+
+<Step label="1">
+
+Install the LTS version of Node.js using Chocolatey:
+
+<Terminal cmd={['$ choco install -y nodejs']} />
+
+If you have already installed Node.js on your system, make sure the version is 18 or above. If you want to be able to switch between different versions, you might want to install Node.js using a version manager such as [`nvm-windows`](https://github.com/coreybutler/nvm-windows).
+
+</Step>
```
```suggestion
```

Same as previous comment about duplicated Node.js info.

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/iosPhysicalDevelopmentBuild.mdx:
```diff
@@ -0,0 +1,81 @@
+<Step label="7">
+### Turn on developer mode
+
+1. Open Settings > Privacy & Security, scroll down to the Developer Mode list item and navigate into it.
+2. Tap the switch to enable Developer Mode. After you do so, Settings presents an alert to warn you that Developer Mode reduces the security of your device. To continue enabling Developer Mode, tap the alert's Restart button.
+3. After the device restarts and you unlock it, the device shows an alert confirming that you want to enable Developer Mode. Tap Turn On, and enter your device passcode when prompted.
```
```suggestion
1. Open "Settings" > "Privacy & Security", scroll down to the "Developer Mode" list item, and navigate into it.
2. Tap the switch to enable "Developer Mode". After you do so, Settings presents an alert to warn you that Developer Mode reduces your device's security. To continue enabling Developer Mode, tap the alert's "Restart" button.
3. After the device restarts and you unlock it, the device shows an alert confirming that you want to enable Developer Mode. Tap "Turn On", and enter your device passcode when prompted.
```

Let's highlight key setting name on the iOS device here (as we do below for scanning QR code in the camera and when the user has to tap on "Open in iTunes".

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/iosPhysicalDevelopmentBuild.mdx:
```diff
@@ -0,0 +1,81 @@
+<Step label="1">
+### Enroll in the Apple Developer Program
+
+To install a development build on your iOS device, you will need an active subscription to the Apple Developer Program. You can sign up for the Apple Developer Program [here](https://developer.apple.com/programs/).
```
```suggestion
To install a development build on your iOS device, you will need an active subscription to the Apple Developer Program. You can sign up for the [Apple Developer Program here](https://developer.apple.com/programs/).
```
Let's highlight the name of the program with the word "here".

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidSimulatedDevelopmentBuild.mdx:
```diff
@@ -0,0 +1,58 @@
+<Step label="5">
+### Install the development build on your emulator
+
+Download the build, then drag and drop the file onto an emulator to install it.
```
```suggestion
Download the build, then drag and drop the file onto an emulator to install it.
```

Should we use similar instructions here as of iOS Simulator: http://docs.expo.dev-pr-28511.s3-website-us-east-1.amazonaws.com/get-started/set-up-your-environment/?platform=ios&device=simulated&mode=development-build#install-the-development-build-on-your-simulator? I think this is the easiest path to just press `Y` and install the build using the CLI.

amandeepmittal commented on docs/pages/get-started/start-developing.mdx:
```diff
@@ -0,0 +1,109 @@
+
+<Collapsible summary="Having problems?">
+
+First, make sure you are on the same Wi-Fi network on your computer and your device.
```
```suggestion
Make sure you are on the same Wi-Fi network on your computer and your device.
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidPhysicalDevelopmentBuildLocal.mdx:
```diff
@@ -0,0 +1,54 @@
+<Step label="1">
+### Enable debugging over USB
+
+Most Android devices can only install and run apps downloaded from Google Play, by default. You will need to enable USB Debugging on your device in order to install your app during development.
```
```suggestion
Most Android devices can only install and run apps downloaded from Google Play, by default. You will need to enable USB Debugging on your device to install your app during development.
```

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidPhysicalDevelopmentBuild.mdx:
```diff
@@ -0,0 +1,54 @@
+To build your app, you will need to create an Expo account and login to the EAS CLI.
+
+1. [Sign up](https://expo.dev/signup) for an Expo account.
+2. Run the following command in your terminal to login to the EAS CLI:
```
```suggestion
2. Run the following command in your terminal to log in to the EAS CLI:
```

- Let's prefer to use the log in as a verb here since this an action

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/androidSimulatedDevelopmentBuild.mdx:
```diff
@@ -0,0 +1,58 @@
+To build your app, you will need to create an Expo account and login to the EAS CLI.
+
+1. [Sign up](https://expo.dev/signup) for an Expo account.
+2. Run the following command in your terminal to login to the EAS CLI:
```
```suggestion
2. Run the following command in your terminal to log in to the EAS CLI:
```

amandeepmittal commented on docs/pages/get-started/start-developing.mdx:
```diff
@@ -0,0 +1,109 @@
+<Step label="2">
+## Open the app on your device
+
+After running the command above, you will see a QR code in your terminal. You can scan this QR code to open the app on your device.
```
```suggestion
After running the command above, you will see a QR code in your terminal. Scan this QR code to open the app on your device.
```

- Let's be a bit more instructive here.

amandeepmittal commented on docs/pages/get-started/start-developing.mdx:
```diff
@@ -0,0 +1,109 @@
+
+<Terminal cmd={['$ npx expo start --tunnel']} />
+
+> Using the **Tunnel** connection type will make the app reloads considerably slower than on **LAN** or **Local**, so it's best to avoid tunnel when possible. You may want to install an emulator/simulator to speed up development if **Tunnel** is required for accessing your machine from another device on your network.
```
```suggestion
> Using the **Tunnel** connection type will make the app reloads considerably slower than on **LAN** or **Local**, so it's best to avoid tunnel when possible. You may want to install and use an emulator or simulator to speed up development if **Tunnel** is required to access your machine from another device on your network.
```

amandeepmittal commented on docs/scenes/get-started/start-developing/ProjectStructure/files/babel-config-js.mdx:
```diff
@@ -0,0 +1,5 @@
+
+<RawH3>babel.config.js</RawH3>
+
+Applies the [babel-preset-expo](https://www.npmjs.com/package/babel-preset-expo) preset that extends the default React Native preset and adds support for decorators, tree-shaking web packages, and loading font icons with optional native dependencies if they're installed. You can also modify this file to add additional Babel plugins or presets.
```
```suggestion
Applies the [`babel-preset-expo`](https://www.npmjs.com/package/babel-preset-expo) preset that extends the default React Native preset and adds support for decorators, tree-shaking web packages, and loading font icons with optional native dependencies if they're installed. You can also modify this file to add additional Babel plugins or presets.
```

- Let's inline code syntax to highlight the package name.

amandeepmittal commented on docs/scenes/get-started/start-developing/ProjectStructure/files/package-json.mdx:
```diff
@@ -0,0 +1,5 @@
+
+<RawH3>package.json</RawH3>
+
+The **package.json** file contains the project's dependencies, scripts and metadata. Anytime a new dependency is added to your project, it will be added to this file.
```
```suggestion
The **package.json** file contains the project's dependencies, scripts, and metadata. Anytime a new dependency is added to your project, it will be added to this file.
```

amandeepmittal commented on docs/scenes/get-started/start-developing/TemplateFeatures/features/themes.mdx:
```diff
@@ -0,0 +1,5 @@
+
+<RawH3>Light and dark modes</RawH3>
+
+This template has light and dark mode support. The `useColorScheme()` hook lets you inspect what the user's current color scheme is, so that you can adjust UI colors accordingly.
```
```suggestion
This template has light and dark mode support. The `useColorScheme()` hook lets you inspect the user's current color scheme so that you can adjust UI colors accordingly.
```

jonsamp commented on docs/pages/get-started/create-a-project.mdx:
```diff
@@ -1,117 +1,22 @@
-  ]}
-  cmdCopy="npx create-expo-app my-app && cd my-app"
-/>
+> You can use another template by adding the `--template` flag.
```
I am rewording to this:

```
> You can select a different template by adding the `--template` flag.
```

- Implicitly, different templates are available. I don't think we need to say "available"
- I liked "select" from your rendition, so I took that.
- I said "different" instead of "from other"

With that, please lmk if you disagree and want another wording.

amandeepmittal commented on docs/scenes/get-started/set-up-your-environment/instructions/_androidStudioEnvironmentInstructions.mdx:
```diff
@@ -0,0 +1,147 @@
+
+</Tab>
+
+<Tab label="SDK 49 and below">
```
We can remove SDK 49 and below instructions since this whole section to the current SDK. After that, we'll also don't have to use `Tabs` here. Can just include Java17 instructions.

Similar changes can be done for Windows down below.

lukmccall committed:
[device][Android] Replace deprecated method to get rotation (#28627)

# Why

Replaces deprecated method to obtain the window rotation.

# Test Plan

- NCL ✅

lukmccall committed:
[video][Android] Remove Kotlin warnings (#28628)

# Why

Removes Kotlin warnings. 
Most of them can be just ignored because we already migrated to new APIs.

# Test Plan

- NCL ✅

lukmccall committed:
[core][Android] Add the method name to unexpected exception (#28626)

lukmccall committed:
[core][Android] Remove Kotlin warnings (#28629)

aleqsio committed:
Add additional config flag to allow forcing RTL on LTR locales (#28129)

# Why and how

There's this issue that has been going on for some time 😅 
https://github.com/expo/expo/issues/26532

It's due to two things:
- first, the plugin on iOS still sets config keys even if there's no
extra config – this is a bug that this PR fixes.
- second, some apps are targeted for RTL markets and require an
additional option to always force RTL, and our locale detection messes
with that – I added a new field to the extra manifest field and config
plugin options.

New key is:

```
app.json > expo > extra > forcesRTL
```

# Test Plan

Performed the following tests:

| | prebuild app | expo go |

|---------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ios | if none set:<br><br>LTR/RTL depend on calling
I18nManager.(force/allow)RTL – built in react-native behavior<br><br>if
extra.forcesRTL true:<br><br>app RTL, I18nManager.isRTL true<br><br>if
extra.supportsRTL true:<br><br>LTR/RTL depend on locale | if none
set:<br><br>app LTR, I18nManager.isRTL false<br><br>if extra.forcesRTL
true:<br><br>app RTL, I18nManager.isRTL true<br><br>if extra.supportsRTL
true:<br><br>LTR/RTL depend on locale |
| android | if none set:<br><br>LTR/RTL depend on calling
I18nManager.(force/allow)RTL – built in react-native behavior<br><br>if
extra.forcesRTL true:<br><br>app RTL, I18nManager.isRTL true<br><br>if
extra.supportsRTL true:<br><br>LTR/RTL depend on locale | if none
set:<br><br>app LTR, I18nManager.isRTL false<br><br>if extra.forcesRTL
true:<br><br>app RTL, I18nManager.isRTL true<br><br>if extra.supportsRTL
true:<br><br>LTR/RTL depend on locale |

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

---------

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

byCedric committed:
feature(expo-build-properties): add `ios.ccacheEnabled` option to enable c++ compiler cache (#28638)

# Why

Closes #28398

This is a re-implementation of #28398, but implemented through
`expo-build-properties`.

# How

- [x] Added `ios.ccacheEnabled` to `expo-build-properties`, to add the
`apple.ccacheEnabled` podfile property.
- [x] Updated the template to enable `:ccache_enabled` once podfile
property is true.
- [x] Update the docs to document this new property.

# Test Plan

See added test

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

---------

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>
Co-authored-by: Kudo Chien <kudo@expo.dev>

EvanBacon committed:
chore(cli): bump canary to support React 19 beta (#28592)

# Why

Support React 19 beta with the following resolutions:

```json
{
    "@types/react": "18.3.1",
    "@types/react-dom": "18.3.0",
    "react": "19.0.0-beta-4508873393-20240430",
    "react-dom": "19.0.0-beta-4508873393-20240430"
}
```

## Test Plan

- ~~Web works as expected but requires a minor change in Router, will
follow up in another PR.~~ landing fix in this PR to keep CI green.
- ~~Native only appears to work with fabric enabled. Setting
`disableLegacyMode = false` and trying again.~~ `disableLegacyMode =
false` works with legacy arch, e.g. Expo Go.

---------

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

brentvatne committed:
[build-properties] Add field to enable/disable privacy manifest aggregation (#28646)

# Why

There are a few issues right now with privacy manifest aggregation
(https://github.com/facebook/react-native/pull/44410,
https://github.com/facebook/react-native/pull/44402,
https://github.com/facebook/react-native/pull/44402) that I expect will
be fixed in React Native 0.74.2, but in order to ensure that this
doesn't break builds I've disabled it by default and added this build
property to allow folks to toggle it on. We can switch on by default
when 0.74.2 is released and proven to handle some edge cases better.

# How

Same old process for adding a new property

# Test Plan

Run tests

# Checklist

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

brentvatne committed:
[packages] Commit build files

brentvatne committed:
[template] Add missing comma

brentvatne committed:
[template] Add privacy_file_aggregation_enabled to the correct place

brentvatne committed:
[docs] Fix Image blurhash syntax in placeholder prop

brentvatne committed:
Publish packages

@expo/cli@0.18.9
expo-build-properties@0.12.1
expo-image@1.12.8
expo-localization@15.0.3
expo-manifests@0.14.2
expo-modules-core@1.12.9
expo-router@3.5.9
expo-sensors@13.0.5
expo-video@1.1.7

brentvatne committed:
Publish packages

expo@51.0.0

brentvatne committed:
[templates] Update to latest [skip ci]

douglowder committed:
Fix TV compile CI (#28648)

Bump dependencies in the CI setup file

brentvatne committed:
[docs] Clarify background location limitations

brentvatne committed:
[changelog] Cut SDK 51 [skip ci]

Simek commented on docs/components/page-higher-order/DocumentationElements.tsx:
```diff
@@ -21,6 +21,10 @@ export default function DocumentationElements(props: DocumentationElementsProps)
     headings: props.headings,
   });
 
+  if (!props.meta) {
+    return props.children;
+  }
+
```
I guess it's a workaround which hides the root error, in general, all pages should have the `meta` prop set.

Can you explain why those lines ware needed?

jonsamp replied:
I added MDX docs that are not full documents, and are instead fragments used inside of UI components. These fragments don't need meta data or any of the providers that full docs do, so I added this exception.

Simek commented on docs/scenes/get-started/set-up-your-environment/PlatformAndDeviceForm.tsx:
```diff
@@ -0,0 +1,87 @@
+export function PlatformAndDeviceForm() {
+  const router = useRouter();
+  const { query, isReady } = router;
+  const [platform, setPlatform] = useState<'android' | 'ios' | null>(null);
+  const [device, setDevice] = useState<'physical' | 'simulated' | null>(null);
```
Same as above, let's make those a separately defined types.

Simek commented on docs/scenes/get-started/set-up-your-environment/SelectCard.tsx:
```diff
@@ -0,0 +1,76 @@
+
+  return (
+    <ButtonBase onClick={onClick}>
+      <div className="w-[250px] shadow-xs overflow-hidden rounded-lg flex flex-col items-center border border-default hocus:scale-[102%] transition-all hocus:shadow-sm">
+        <div
```
```suggestion
      <div className={mergeClasses(
        "w-[250px] shadow-xs overflow-hidden rounded-lg flex flex-col items-center border border-default transition-all", 
        "hocus:scale-[102%] hocus:shadow-sm"
      )}>
        <div
```

It's a good practice to split TW class name strings for the readability, we usually use `mergeClasses` helper util for that, and split the classes by scope or state.

Simek commented on docs/scenes/get-started/set-up-your-environment/SelectCard.tsx:
```diff
@@ -0,0 +1,76 @@
+  const { themeName } = useTheme();
+  const [isDarkMode, setIsDarkMode] = useState(false);
+
+  useEffect(
+    function didMount() {
+      const darkMode =
+        themeName === 'dark' ||
+        (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches);
+      setIsDarkMode(darkMode);
+    },
+    [themeName]
+  );
```
This code should be redundant, we have a global theme provider and management:
```jsx
import { useTheme } from '@expo/styleguide';
```

Simek replied:
This code also leads to always loading light mode images, no matter of the theme, which leads to flicker on load when user use the dark mode:

![xx](https://github.com/expo/expo/assets/719641/258f4f0d-4907-46eb-a880-4366dce823f3)

jonsamp replied:
I'll fix this, thanks for the suggestion!

jonsamp replied:
Oh right, my problem was that if you have "auto" set, then the `themeName` is not an accurate way to tell the theme, since then we need to listen for `prefers-color-scheme: dark`. I will remove that condition right now since it's causing that flicker but I can update our `useTheme()` hook later to support auto mode.

Simek commented on docs/scenes/get-started/set-up-your-environment/SelectCard.tsx:
```diff
@@ -0,0 +1,76 @@
+          )}>
+          <picture className="relative w-auto h-full flex items-end">
+            {isDarkMode && <source srcSet={darkImgSrc} type="image/png" />}
+            <img
+              src={imgSrc}
+              alt={alt}
+              className={mergeClasses(isSelected ? 'grayscale-0' : 'grayscale opacity-80')}
+            />
```
Let's set the image sizes, or at least min-height to the container, so we can avoid lout shifts on render.

Simek commented on docs/scenes/get-started/set-up-your-environment/SelectCard.tsx:
```diff
@@ -0,0 +1,76 @@
+        <div className="p-3 text-center flex items-center justify-center flex-col gap-2">
+          <HEADLINE className="text-center">{title}</HEADLINE>
+          {description ? <CALLOUT theme="secondary">{description}</CALLOUT> : null}
+          <div className="relative">
+            <div
+              className={mergeClasses(
+                'size-5 rounded-full border bg-default',
+                isSelected ? 'border-palette-blue9' : 'border-default'
+              )}
+            />
+            <div
+              className={mergeClasses(
+                'size-3 rounded-full absolute top-1 right-1',
+                isSelected ? 'border-palette-blue9 bg-palette-blue9' : 'bg-transparent'
+              )}
+            />
+          </div>
```
Let's move in `Radio` input component, instead of trying to recreate only the visual part of it. 

I really don't like the radio placement and alignment of the tiles content. Can we move the input indicator to the header line and stick to the text aligned to the left? Also the interaction animation is not exactly what we usually use, would be nice if we can move scale to only apply to asset used in tile, not the whole tile.

<img width="698" alt="Screenshot 2024-05-07 103250" src="https://github.com/expo/expo/assets/719641/140ad8c0-e048-4202-9315-3fa3f8e8405c">

jonsamp replied:
I am currently about to be on a plane, and this needs to go out today. With that, I can address this in about 12 hours in another PR. I understand that we want to change the design/functionality here. I will get that PR up later today.

Simek reviewed (changes requested):
See comments above.

lukmccall committed:
[docs] Remove incorrect usage of `launch` function (#28634)

tsapeta committed:
[video] Clean up types and export the VideoPlayer class (#28623)

alanjhughes committed:
[camera] Support toggling `autoFocus` on and off (#28650)

byCedric committed:
docs: drop additional references to `expo-cli` from docs (#28613)

# Why

We still had this.

# How

Dropped additional references to the classic `expo-cli` from docs.

# Test Plan

Docs change

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

amandeepmittal committed:
[docs] Update Upgrade Expo SDK (#28457)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Closes ENG-11148

# How

<!--
How did you build this feature or fix this bug and why?
-->

- By updating Expo Go context to indicate that only a single version
(latest SDK version) is supported. Also, add expo.dev/go information to
allow installing a specific Expo Go version.
- Increment Expo package version in Terminal component to SDK 50.


# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run docs locally and visit:
http://localhost:3002/workflow/upgrading-expo-sdk-walkthrough/

## Preview

![CleanShot 2024-04-25 at 21 27
01@2x](https://github.com/expo/expo/assets/10234615/96583cae-7df5-405a-804e-28b4f2541e8f)

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

amandeepmittal committed:
[docs] Update outdated info in Expo SDK libraries section (#28504)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->
This PR:
- Platform compatibility table with platform tags and a screenshot taken
from `expo-device` as an example.
- Updates info about what an API reference page contains to include the
config plugin.
- Remove `expo-device` example as it has no relevance here (as a reader
I want to learn what an API reference page contains)
- Update styles for React Native components API example.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run docs locally and visit:
http://localhost:3002/workflow/using-libraries/#expo-sdk-libraries

## Preview

![CleanShot 2024-04-29 at 00 47
37@2x](https://github.com/expo/expo/assets/10234615/16caaeb4-c0fa-4685-9887-94839e71c6ac)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

brentvatne committed:
[create-expo] Default to expo-template-default

EvanBacon committed:
add basic e2e tests for Metro web Fast Refresh (#28237)

# Why

- It's important we keep Metro web working continuously. The behavior of
Fast Refresh is a bit of a black box, so having a test gives us a
clearer list of expected steps that must work.

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

# How

- I've added Playwright to the CLI and an additional script for running
playwright tests.
- The CI job is split out of the E2E CLI job so we can get faster
failures. Pinging @byCedric for advice on the structure and how we could
potentially reuse work between these jobs.

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

- Running `yarn test:playwright` in `packages/@expo/cli` works as
expected locally.
- CI should run the playwright test continuously. We'll also be keeping
an eye on the timings to ensure this test runs fast and doesn't become
another burden on our merge process.

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

gabrieldonadel committed:
[cli] Filter out unavailable connected devices (#28642)

# Why

While testing `devicectl list devices` for Orbit I noticed that xcode
returns iOS devices even if the device is unavailable (E.g. wifi and
Bluetooth off).

Upon testing this with Expo CLI I was faced with the following issues:

1. the iPhone was shown as connected via cable


![image](https://github.com/expo/expo/assets/11707729/e38102f1-9969-4371-87a2-cfdd8b5b5181)

2. When selecting the device the build would fail with `Unable to find a
destination matching the provided destination`


![image](https://github.com/expo/expo/assets/11707729/66240243-9aef-4f0f-9907-3a8c8ff539fe)


# How

Update `getConnectedDevicesUsingNativeToolsAsync` to check the tunnel
state of the device and filter out unavailable devices.

# Test Plan

```
npx expo run:ios -d
```

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

tsapeta committed:
Remove patch for react-native-reanimated in ReanimatedUIManager (#28651)

amandeepmittal committed:
[docs] Update ESLint guide to include info on `npx expo lint` (#28630)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Add info about `npx expo lint` in ESLint guide for SDK 51 and above.

# How

<!--
How did you build this feature or fix this bug and why?
-->

By updating sections under ESLint and Prettier to include the new
information and re-structure the existing information accordingly.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

Run docs locally and visit:
http://localhost:3002/guides/using-eslint/#eslint

## Preview
![CleanShot 2024-05-06 at 03 10
17@2x](https://github.com/expo/expo/assets/10234615/2c7f3e01-52b0-42fd-8a44-09dddab3e9a1)

![CleanShot 2024-05-06 at 03 10
24@2x](https://github.com/expo/expo/assets/10234615/018b7055-516d-4441-8fb2-55b4ed5d4f1c)



# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

brentvatne committed:
[docs] Update latest version

brentvatne committed:
Publish packages

@expo/cli@0.18.10
expo-camera@15.0.4
expo-module-template@10.14.7
expo-updates@0.25.9
expo-video@1.1.8

jonsamp committed:
adds boxlink back
